server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -201,6 +204,9 @@ public RemoteClusterStateService(
             threadPool
         );
         this.remoteClusterStateCleanupManager = new RemoteClusterStateCleanupManager(this, clusterService, remoteRoutingTableService);
+        this.isPublicationEnabled = FeatureFlags.isEnabled(REMOTE_PUBLICATION_EXPERIMENTAL)
```

Comment:
this is flag is already present ? can we reuse?

Reply:
That method is there in DiscoveryNode. We need instance of discoveryNode to use here.